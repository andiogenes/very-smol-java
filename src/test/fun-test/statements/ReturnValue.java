class Main {
    // Должен сработать первый return
    int foo() {
        return 1;
        return 2;
        return 3;
        return 4;
    }

    int x;

    // return в условных операторах
    double bar() {
        switch (x) {
            case 0:
                return 1.E0;
            default:
                return 2.E0;
        }
        return 0.E0;
    }

    // приведение типов в return
    double cast() { return 9; }

    void main() {
        int a = foo();

        x = 0;
        double b = bar();

        x = 1;
        double c = bar();

        double d = cast() / 2;
    }
}