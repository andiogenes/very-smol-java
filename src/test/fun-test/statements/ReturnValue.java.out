x : INT = 0 : INT
a : INT = 1 : INT
x : INT = 0 : INT
b : DOUBLE = 1.0 : DOUBLE
x : INT = 1 : INT
c : DOUBLE = 2.0 : DOUBLE
d : DOUBLE = 4.5 : DOUBLE
